updater/lib/dependabot/updater/operations/group_update_all_versions.rb

```diff
@@ -113,32 +112,19 @@ def run_grouped_dependency_updates
               Dependabot.logger.info(
                 "Deferring creation of a new pull request. The existing pull request will update in a separate job."
               )
-
-              # add the dependencies in the group so individual updates don't try to update them
-              dependency_snapshot.add_handled_dependencies(
-                dependencies_in_existing_pr_for_group(group).filter_map { |d| d["dependency-name"] }
-              )
-              # also add dependencies that might be in the group, as a rebase would add them;
-              # this avoids individual PR creation that immediately is superseded by a group PR supersede
-              dependency_snapshot.add_handled_dependencies(group.dependencies.map(&:name))
+              dependency_snapshot.mark_group_handled(group)
               next
             end
 
             group
           end
 
           groups_without_pr.each do |group|
-            grouped_update_result = run_grouped_update_for(group)
-            if grouped_update_result
-              # Add the actual updated dependencies to the handled list so they don't get updated individually.
-              dependency_snapshot.add_handled_dependencies(grouped_update_result.updated_dependencies.map(&:name))
-            else
-              # The update failed, add the suspected dependencies to the handled list so they don't update individually.
-              dependency_snapshot.add_handled_dependencies(group.dependencies.map(&:name))
-            end
+            dependency_change = run_grouped_update_for(group)
+            # The update failed, add the suspected dependencies to the handled list so they don't update individually.
+            dependency_snapshot.mark_group_handled(group) if dependency_change.nil?
```

Comment:
Do the dependencies from a successful dependency_change still get marked as handled here?

Reply:
Great question, this was actually the bug that I found!

During the course of `compile_all_dependency_changes_for` we actually mark dependencies as handled, and in the appropriate directory.

Well here in `group_update_all_versions`, the `current_directory` happened to not be cleared, so it was marking the final directory seen as handled for all dependencies encountered in the update! 

Removing this superfluous call to `add_handled_dependencies` fixes it.